src/content.js

```diff
@@ -0,0 +1,1114 @@
+  if (!event.isTrusted) return;
+  const el = event.target;
+  const cl = el.className;
+
+  let hasRelatedImage = false;
+  if (
+    IsTumblr &&
+    [
+      "post_controls_top",
+      "post_tags_inner",
+      "click_glass",
+      "hover",
+      "hover_inner",
+      "post_date",
+      "post_notes"
+    ].includes(cl)
+  ) {
+    hasRelatedImage = setTumblrRelatedImage(el);
+  }
+
+  if (IsPinterest && (cl === "hoverMask" || cl === "playIndicatorPill")) {
+    handlePinterestHover(el);
+    return;
+  }
+
+  if (CurrentHover) {
+    if (partOfHoverTarget(el)) {
+      CurrentHover.clearTimeouts();
+      return;
+    }
+    clearHoverEffect();
+  }
+
+  if (hasRelatedImage) handleImgHover(el[eRelatedTo]);
+  else if (
+    (el.tagName === "IMG" || isGifv(el)) &&
+    el instanceof window.HTMLElement
+  )
+    handleImgHover(el);
+}
+
+function onMouseOut(event) {
+  if (!event.isTrusted) return;
+  HoveredNonanimatedImage = null;
+  if (!CurrentHover || CurrentHover.mouseoutTimeout !== null) return;
+
+  // Essentially we want to do clearHoverEffect here, since we unhovered an
+  // animated image. However, the mouseout might be because we hovered over
+  // one of the buttons, which we will know in a few milliseconds. Hence this
+  // complicated logic.
+  try {
+    CurrentHover.mouseoutTimeout = window.setTimeout(() => {
+      CurrentHover.mouseoutTimeout = null;
+      clearHoverEffect();
+    }, 50);
+  } catch (ex) {
+    clearHoverEffect();
+  }
+}
+
+function updateHoverListeners() {
+  window.addEventListener("mouseover", onMouseOver);
+  window.addEventListener("mouseout", onMouseOut);
+}
+
+function markAnimated(img) {
+  if (img[eCheckedAnimation]) return;
+  img[eCheckedAnimation] = true;
+
+  if (img[eAwaitingPlay]) setAnimationState(img, "normal");
+  else if (WantedAnimationBehavior !== AnimationBehavior)
+    setAnimationState(img, WantedAnimationBehavior);
+
+  updateIndicator(img);
+
+  if (HoveredNonanimatedImage === img) applyHoverEffect(img);
+}
+
+function handleLoad(el) {
+  const callback = el[eOnLoad];
+  if (callback) {
+    el[eOnLoad] = null;
+    callback();
+    return;
+  }
+
+  if (
+    Prefs.playOnHover &&
+    Prefs.hoverPlayOnLoad &&
+    (Prefs.hoverPauseWhen === HoverPause.Never || LastToggledImage === el) &&
+    el[eHasHovered] &&
+    isAnimatedImage(el) &&
+    getAnimationState(el) === "none"
+  ) {
+    setAnimationState(el, "normal");
+    updateIndicator(el);
+    if (CurrentHover) CurrentHover.refresh();
+  }
+}
+
+function handleLoadedMetadata(el) {
+  if (isGifv(el)) {
+    if (el[eInitedGifv]) return;
+    el[eInitedGifv] = true;
+    if (Prefs.defaultPaused) setAnimationState(el, "none");
+    updateIndicator(el);
+  }
+}
+
+// ==== Animation detection ====
+
+function notifyAnimated(url) {
+  console.info("content-script: found animated image", url);
+  AnimatedMap.set(hash(url), 1);
+  SeenAnyAnimated = true;
+  document.getElementsByTagName("img").forEach(img => {
```

Comment:
oh, right, I also saw this line failing, because `getElementsByTagName` doesn't return an array. I wish lints were better at this sort of thing :(

Reply:
I changed it to `[...document.getElementsByTagName("img")].forEach`